### PR TITLE
More corrections to lua_api.txt

### LINKED ISSUE #12531: `preserve_metadata` passes fields, not a `NodeMetaRef`, for `oldmeta`
##### Minetest version
```
5.6-dev
```

##### Summary

Found by @lukasdragon.

Implementation: https://github.com/minetest/minetest/blob/8724fe6e3fc2b6c0b78123f1f95fd8c6c6817dd5/builtin/game/item.lua#L490

Docs: https://github.com/minetest/minetest/blob/8724fe6e3fc2b6c0b78123f1f95fd8c6c6817dd5/doc/lua_api.txt#L8294

### PR DESCRIPTION
**Pull request by sfan5**
also incorporates #11632
fixes #12531

## To do

This PR is Ready for Review.

### CODE REVIEW

**Comment by rubenwardy** [comment]
What about incorporating #12448 ? It was never merged, the author was going to work on more fixed but never did

File: doc/lua_api.txt [hunk 1]
```diff
@@ -6810,6 +6807,21 @@ Lua back to the engine.
 Doing so is much less error-prone and you will never need to wonder if the
 object you are working with still exists.
 
+### Attachments
+
+It is possible to attach objects to other objects (`set_attach` method).
+
+When an object is attached, it is positioned relative to the parent's position
+and rotation. `get_pos` and `get_rotation` will always return the parent's
+values and can't be set directly.
```
Comment (SmallJoker) [inline doc/lua_api.txt:6816]:
```suggestion
values and changes via their setter counterparts are ignored.
```

File: doc/lua_api.txt [hunk 2]
```diff
@@ -6862,12 +6874,13 @@ object you are working with still exists.
 * `set_animation_frame_speed(frame_speed)`
     * `frame_speed`: number, default: `15.0`
 * `set_attach(parent[, bone, position, rotation, forced_visible])`
-    * `bone`: string. Default is `""`, the root bone
-    * `position`: `{x=num, y=num, z=num}`, relative, default `{x=0, y=0, z=0}`
-    * `rotation`: `{x=num, y=num, z=num}` = Rotation on each axis, in degrees.
-      Default `{x=0, y=0, z=0}`
+    * `parent`: object to attach to
```
Comment (SmallJoker) [inline doc/lua_api.txt:6877]:
```suggestion
    * `parent`: `ObjectRef` to attach to
```

File: doc/lua_api.txt [hunk 3]
```diff
@@ -4508,26 +4508,23 @@ generated chunk by the current mapgen.
...
-nodes, not the decorations themselves. A 'simple' type decoration is often 1
-node above the returned position and possibly displaced by 'place_offset_y'.
-
-Possible fields of the table returned are:
+the current chunk. To enable the capture of positions of interest to be recorded
+call `minetest.set_gen_notify()` first.
 
-* `dungeon`
-* `temple`
+Possible fields of the returned table are:
```
Comment (rubenwardy) [inline doc/lua_api.txt:4514]: Markdown lists require an empty line either side:

```suggestion
Possible fields of the returned table are:

```

File: doc/lua_api.txt [hunk 4]
```diff
@@ -4508,26 +4508,23 @@ generated chunk by the current mapgen.
...
 * `large_cave_begin`
 * `large_cave_end`
-* `decoration`
+* `decoration#id` (see below)
 
 Decorations have a key in the format of `"decoration#id"`, where `id` is the
-numeric unique decoration ID as returned by `minetest.get_decoration_id`.
-
-
+numeric unique decoration ID as returned by `minetest.get_decoration_id()`.
```
Comment (rubenwardy) [inline doc/lua_api.txt:4524]: example + new paragraph

```suggestion
numeric unique decoration ID as returned by `minetest.get_decoration_id()`.
For example, `decoration#123`.

```

File: doc/lua_api.txt [hunk 5]
```diff
@@ -6810,6 +6807,21 @@ Lua back to the engine.
...
+It is possible to attach objects to other objects (`set_attach` method).
+
+When an object is attached, it is positioned relative to the parent's position
+and rotation. `get_pos` and `get_rotation` will always return the parent's
+values and changes via their setter counterparts are ignored.
+
+To change position or rotation call `set_attach` again with the new values.
+
+**Note**: Just like model dimensions, the relative position in `set_attach`
+must be multiplied by 10.
```
Comment (rubenwardy) [inline doc/lua_api.txt:6821]:
```suggestion
must be multiplied by 10 compared to world positions.
```

**Review by rubenwardy — APPROVED** [review]

File: doc/lua_api.txt [hunk 6]
```diff
@@ -8890,30 +8902,30 @@ Used by `minetest.create_detached_inventory`.
 HUD Definition
 --------------
 
-See [HUD] section.
+Since most values have multiple different functions, please see the
+documentation in [HUD] section.
 
-Used by `Player:hud_add`. Returned by `Player:hud_get`.
+Used by `ObjectRef:hud_add()`. Returned by `ObjectRef:hud_get()`.
```
[the remark this answers is not in the document]
Comment (SmallJoker) [reply doc/lua_api.txt]: Yes.

**Pull request merged**